tests/app/questionnaire/test_completeness.py

```diff
@@ -0,0 +1,483 @@
+            expected_location,
+        ]
+
+        progress = Completeness(
+            schema, AnswerStore(), completed_blocks, routing_path, metadata={})
+
+        invalid_location = progress.get_first_incomplete_location_in_survey()
+        self.assertEqual(expected_location, invalid_location)
+
+    def test_get_first_incomplete_location_in_survey_started_with_repeat(self):
+        schema = load_schema_from_params('test', 'navigation')
+
+        completed_blocks = [
+            Location('multiple-questions-group', 0, 'household-composition'),
+            Location('extra-cover', 1, 'extra-cover-block'),
+            Location('repeating-group', 0, 'repeating-block-1'),
+            Location('repeating-group', 1, 'repeating-block-1'),
+            Location('extra-cover', 0, 'extra-cover-block'),
+            Location('extra-cover-items-group', 0, 'extra-cover-items'),
+        ]
+
+        expected_location = Location('extra-cover-items-group', 1, 'extra-cover-items')
+        routing_path = [
+            Location('multiple-questions-group', 0, 'household-composition'),
+            Location('repeating-group', 0, 'repeating-block-1'),
+            Location('repeating-group', 1, 'repeating-block-1'),
+            Location('extra-cover', 0, 'extra-cover-block'),
+            Location('extra-cover-items-group', 0, 'extra-cover-items'),
+            expected_location,
+        ]
+
+        answer_store = AnswerStore()
+        answer_store.add(Answer(
+            answer_instance=0,
+            answer_id='first-name',
+            value='Person1'
+        ))
+        answer_store.add(Answer(
+            answer_instance=1,
+            answer_id='first-name',
+            value='Person2'
+        ))
+        answer_store.add(Answer(
+            answer_instance=0,
+            answer_id='extra-cover-answer',
+            value=2
+        ))
+
+        progress = Completeness(
+            schema, answer_store, completed_blocks, routing_path, metadata={})
+
+        invalid_location = progress.get_first_incomplete_location_in_survey()
+        self.assertEqual(expected_location, invalid_location)
+
+    def test_get_first_incomplete_location_in_survey_completed(self):
+        schema = load_schema_from_params('test', 'navigation_completeness')
+        completed_blocks = [
+            Location('coffee-group', 0, 'coffee'),
+            Location('coffee-group', 0, 'response-yes'),
+            Location('toast-group', 0, 'toast'),
+        ]
+        routing_path = [
+            Location('coffee-group', 0, 'coffee'),
+            Location('coffee-group', 0, 'response-yes'),
+            Location('toast-group', 0, 'toast'),
+        ]
+
+        progress = Completeness(
+            schema, AnswerStore(), completed_blocks, routing_path, metadata={})
+        self.assertFalse(progress.get_first_incomplete_location_in_survey())
+
+    def get_first_incomplete_location_in_survey_completed_skipped_interstitials(self):
```

Comment:
doesn't this test also test everything tested in the test above?

Reply:
Yes, true.